### PR TITLE
Added save files patcher

### PR DESCRIPTION
**Pull request by lciesielski**
Added new logic that will try to find Diablo 2 Resurrected save files folder and patch saves as if characters have finished normal difficulty.
This allows to play offline (without TCP/IP).
Also as a safety measure backups are created just in case.

### CODE REVIEW

**Comment by mfrklic** [comment]
I like it. Only suggestion I have is to be able to disable it via param or make it so it only runs via param where you can pass a character name or do a wildcard like '*':

example:
d2roffline.exe -FixLocalSave MyCharacterName
d2roffline.exe -FixLocalSave *

or to disable:
d2roffline.exe -SkipLocalSaveFix

The only reason I would suggest to not run this every time is #1 in case it causes delays with the timers and/or if like me I have characters that are in Hell that I do not want to get reset to Nightmare for example.

But agree that it is nice to have it automatically resolve issues with saves for most users.

**Comment by lciesielski** [comment]
@mfrklic that's a very good suggestion, users would be rather upset if they lost progression, pull updated

**Comment by ferib** [comment]
> example:
d2roffline.exe -FixLocalSave MyCharacterName
d2roffline.exe -FixLocalSave *
> or to disable:
d2roffline.exe -SkipLocalSaveFix

That sounds like a great idea!

> The only reason I would suggest to not run this every time is #1 in case it causes delays with the timers and/or if like me I have characters that are in Hell that I do not want to get reset to Nightmare for example.
But agree that it is nice to have it automatically resolve issues with saves for most users.

It is indeed nice to have this auto-resolve for other users, but I think it would be best to make it so the game won't run when you call the `FixLocalSave`? so it will only Fix the local saves, and if the user wants to start the game, it will just use the tool as normal.

**Comment by lciesielski** [comment]
@ferib sure, I thought that AIO package would be simpler for most users, however your idea has the upper hand optimization wise

File: Program.cs [hunk 1]
```diff
@@ -21,8 +22,18 @@ static void Main(string[] args)
             Console.ForegroundColor = ConsoleColor.Gray;
 
             // overwrite path if args are set
-            if (args.Length > 0)
-                d2rPath = args[0];
+            if (args.Length > 0)
+            {
+                if (args[0].Equals("-FixLocalSave", StringComparison.InvariantCultureIgnoreCase))
```
Comment (lciesielski) [inline Program.cs:27]: Wanted to use switch here, but switch does not support ignore case ...
Could got with ToUpper or ToLower but it's not optimal

**Comment by ferib** [comment]
For some reason, my VS crashes whenever I open the project, other than that, everything looks good!

**Comment by ferib** [comment]
FYI please use the CR LR line endings, I think most ppl here use windows which uses CR LF by default, thanks.

**Comment by fuba82** [comment]
hmm after the "-FixLocalSave" command makes my "Level 1" Character is able to go to Hell difficulty?! ^^

**Pull request merged**